src/ledger/LedgerManagerImpl.cpp

```diff
@@ -386,6 +392,8 @@ LedgerManagerImpl::loadLastKnownLedger(function<void()> handler)
         {
             LedgerTxn ltx(mApp.getLedgerTxnRoot());
             advanceLedgerPointers(ltx.loadHeader().current());
+            // Ledger is not ready in this flow yet, so we can't update
+            // the network config.
```

Comment:
This portion of startup is used for when we don't run the app, but still create it to perform some actions on its state. There are some commands that use loadLastKnownLedger without the handler, so not loading the network config potentially breaks those. I think it should be safe to load the config from the DB here

Reply:
This comment is based on my vague understanding of this flow and it was inferred from de-facto behavior of of the code as exercised by the tests. IIRC I just couldn't load the config here. Looking at the difference between the branches you can see that e.g. we may not even have buckets here. I honestly don't understand why do we need this diversion at all instead of just not calling the handler when it's missing; but it seems like the handler presence just indicates more fundamental difference...

If you have an idea of what flow could be affected by this, but not be covered in the current tests, then we should probably cover it and make sure it does (or doesn't) work.